### ISSUE TITLE
Does this support declaration merging?

### ISSUE THREAD

**Issue by bywo** [timocov/dts-bundle-generator#148]
I tried it once with my codebase and it didn't work as expected, but I'm not sure if it's a bug or if I'm not using the tool correctly.

Here's my scenario:

```ts
// mergedType.ts
export interface MergedType {}
```

```ts
// a.ts
declare module "./mergedType.ts" {
  interface MergedType {
    a: boolean;
  }
}
```

```ts
// b.ts
declare module "./mergedType.ts" {
  interface MergedType {
    b: boolean;
  }
}
```

I'm hoping that `dts-bundle-generator` can emit something like
```ts
export interface MergedType {
  a: boolean;
  b: boolean;
}
```

or

```ts
export interface MergedType {}
interface MergedType {
  a: boolean;
}
interface MergedType {
  b: boolean;
}
```

I can put together a demo repo as well if that helps. Thanks!

**Comment by timocov** [comment]
Yes, please. Also, you can takea look at this test case https://www.github.com/timocov/dts-bundle-generator/tree/master/tests%2Fe2e%2Ftest-cases%2Fextend-other-module, where I tested exactly your case and it seems that it works as you expect.

**Comment by bywo** [comment]
I was able to create a repro case: https://github.com/timocov/dts-bundle-generator/pull/150

**Comment by timocov** [comment]
@bywo Thanks! I'm able to reproduce issue now. This is a bug, I'll fix it soon.

**Comment by timocov** [comment]
Fixed in #150

**Comment by timocov** [comment]
The fix has been published in 5.8.0 version. Also, please take a look at this note https://github.com/timocov/dts-bundle-generator/pull/150#discussion_r598099780.
